Day 1/DayOnePartOne.py

```diff
@@ -0,0 +1,36 @@
+import sys
+sys.path.insert(0, "C:/Users/Callipolis/Desktop/Programozás/AoC2/Advent-of-Code-2015")
+from InputReader import InputReader
+
+input_file = InputReader("input1.txt")
+raw_input = input_file.get_raw_input()
+
+class ParenthesisNavigator:
+    
+    def __init__(self, puzzle_input):
+        self.puzzle_input = puzzle_input
+        self.floor_counter = 0
+    
+    def __str__(self):
+        return "The solution is: " + str(self.floor_counter)
+    
+    def change_floor(self, move):
+        if move == "(":
```

Comment:
Ezzel Tesot is oruletbe kergettem, de az input parse-olasa es a feladat emgoldasa szet kell legyen valasztva. Ha adok neked egy file-t ahol "f" jeloli a felt es "l" a le-t zarojelek helyett akkor hozza kell nyulnod ehhez a class-hoz, holott semmi relevans nem valtozott igazabol. 
Kicsit ez alapjan is gondolhatsz responsibility-kre: "How many different things can change in the problem statement that makes me change this class"

Itt most a responsibilty az az hogy kap egy listat/iterator-t aminek az elemei egyertelmuen indikaljak hogy fel vagy le ment es az alapjan megmondja a floor szamot. Nincs UI, formazas, parse-olas, egyeb. 

Fix:
ket opciod van (biztos tobb is, de most ketto obvious-t latok)
- Parse-olast kulon szeded es atalakitod az input bemenetet es zarojelek helyet True/False-t hasznalsz vagy enum-ot, vagy legalabb 1/0-t (ne). Az atalakitott list-et adod at ennek a class-nak.
- Ha nem akarod atalakitani (mert esetleg olyan hosszu az input hogy megterhelne ez a memoriat, akkor meg szedd ki a class-bol azt a logikat ami eldonti hogy a zarojelek mit jelentenek. Az inputtal egyutt atadhatsz egy fuggvenyt mondjuk ami egy bemeno karakterre visszaadja azt hogy fel vagy le ment az ember. Ha valtozik az input akkor egy masik fuggvenyt adsz at ennek a class-nak.

En az elsot valasztanam es a legtobb esetben az a jo megkozelites. Arulkodo jel a print "Input character not recognized" is. Azt hogy szar az input az elott kellene megallapitanod hogy barmit is csinalsz (fail fast). Az hogy felig process-elted mar a dolgokat es akkor jossz ra hogy szar az input-od az keso. 
Ha eloszor parse-olod es utana a modositott input-ot adod at akkor elore kiderul parse-olas soran ha szar az input es tudsz szolni a user-nek mielott barmi szamitast elkezdenel.

Reply:
Áh értem, minél jobban szétbontod a feladatokat, annál könnyebb a javítás és gyorsabb a hibafellelés. Atom, szét is bontottam.

```
    def parse_input(self):
        parsed_instructions = []
        
        for character in self.instructions:
            if character == self.up_char:
                parsed_instructions.append(True)
            elif character == self.down_char:
                parsed_instructions.append(False)
            else:
                raise ValueError(f"Input character not recognized: {character}")

        return parsed_instructions
```